File: libs/utils/lib-core/src/main/java/com/yyxnb/what/core/log/IPrinter.java
package com.yyxnb.what.core.log;

import java.util.List;
import java.util.Map;

public interface IPrinter {

    LogConfig init();

    String getFormatLog();

    void d(String message, Object... args);

    void e(String message, Object... args);

    void e(Throwable throwable, String message, Object... args);

    void w(String message, Object... args);

    void i(String message, Object... args);

    void v(String message, Object... args);

    void wtf(String message, Object... args);

    void json(String json);

    void xml(String xml);

    void map(Map map);

    void list(List list);
}

File: arch/src/main/java/com/yyxnb/arch/interfaces/IPrinter.kt
package com.yyxnb.arch.interfaces

import com.yyxnb.arch.utils.log.LogConfig

interface IPrinter {

    val formatLog: String

    fun init(): LogConfig

    fun d(message: String, vararg args: Any)

    fun e(message: String, vararg args: Any)

    fun e(throwable: Throwable?, message: String?, vararg args: Any)

    fun w(message: String, vararg args: Any)

    fun i(message: String, vararg args: Any)

    fun v(message: String, vararg args: Any)

    fun wtf(message: String, vararg args: Any)

    fun json(json: String)

    fun xml(xml: String)

    fun map(map: Map<*, *>?)

    fun list(list: List<*>?)
}
